File: Leetcode 1301-1400/1342/1342.py
# Leetcode 1342: Number of Steps to Reduce a Number to Zero
# https://leetcode.com/problems/number-of-steps-to-reduce-a-number-to-zero/
# Solved on 27th of February, 2026
class Solution:
    def numberOfSteps(self, num: int) -> int:
        """
        Calculates the number of steps to reduce a non-negative integer to zero.
        :param num: The integer to be reduced.
        :return: The total number of steps taken.
        """
        stepCount = 0

        while num > 0:
            if num & 1:
                num -= 1
            else:
                num >>= 1
            stepCount += 1

        return stepCount

File: Leetcode 1301-1400/1342/1342.java
// Leetcode 1342: Number of Steps to Reduce a Number to Zero
// https://leetcode.com/problems/number-of-steps-to-reduce-a-number-to-zero/
// Solved on 27th of February, 2026
class Solution {
    /**
     * Calculates the number of steps to reduce a non-negative integer to zero.
     * If the current number is even, it is divided by 2; otherwise, 1 is subtracted from it.
     * 
     * @param num The non-negative integer to reduce.
     * @return The total number of steps taken to reach zero.
     */
    public int numberOfSteps(int num) {
        int steps = 0;
        while (num > 0) {
            if ((num & 1) == ) {
                num >>= 1;
            } else {
                num--;
            }
            steps++;
        }
        return steps;
    }
}